pkg/cli/session/session_test.go

```diff
@@ -1 +1,212 @@
+
+			Expect(err).To(BeNil())
+
+			// Check history file
+			historyFile, err := os.Stat(filepath.Join(bpSession.Path, ".history"))
+			Expect(err).To(BeNil())
+			Expect(historyFile).NotTo(BeNil())
+
+			// check bash env file
+			ocEnvFile, err := os.Stat(filepath.Join(bpSession.Path, ".ocenv"))
+			Expect(err).To(BeNil())
+			Expect(ocEnvFile).NotTo(BeNil())
+
+			// check zsh env file
+			zshEnvFile, err := os.Stat(filepath.Join(bpSession.Path, ".zshenv"))
+			Expect(err).To(BeNil())
+			Expect(zshEnvFile).NotTo(BeNil())
+		})
+
+		It("Check backplane session folder permissions", func() {
+			mockOcmInterface.EXPECT().GetTargetCluster(options.Alias).Return("", "", errors.New("err")).AnyTimes()
+			err := bpSession.Setup()
+
+			Expect(err).To(BeNil())
+
+			// Write Kube config file to session folder
+			kubeConfigWrite, err := os.Create(filepath.Join(bpSession.Path, "config"))
+			Expect(err).To(BeNil())
+			Expect(kubeConfigWrite).NotTo(BeNil())
+
+			// Read Kube config file from session folder
+			kubeConfigRead, err := os.Stat(filepath.Join(bpSession.Path, "config"))
+			Expect(err).To(BeNil())
+			Expect(kubeConfigRead).NotTo(BeNil())
+
+		})
+	})
+
+	Context("test Backplane Run Command", func() {
+		It("should fail for invalid cluster alias name ", func() {
+			options.Alias = "my-session"
+
+			mockOcmInterface.EXPECT().GetTargetCluster(options.Alias).Return("", "", errors.New("err")).AnyTimes()
+
+			err := bpSession.RunCommand(cmd, []string{})
+			Expect(err).NotTo(BeNil())
+			Expect(err.Error()).Should(ContainSubstring("invalid cluster Id my-session"))
+		})
+
+		It("should fail for empty session name ", func() {
+			options.Alias = ""
+			options.ClusterId = ""
+
+			err := bpSession.RunCommand(cmd, []string{})
+			Expect(err).NotTo(BeNil())
+			Expect(err.Error()).Should(ContainSubstring("ClusterId or Alias required"))
+		})
+
+		It("should use clusterID when alias is empty ", func() {
+			options.Alias = ""
+			options.ClusterId = testClusterId
+
+			mockClientWithResp.EXPECT().LoginClusterWithResponse(gomock.Any(), gomock.Any()).Return(nil, nil).AnyTimes()
+			mockOcmInterface.EXPECT().GetTargetCluster(options.ClusterId).Return(trueClusterId, testClusterId, nil).AnyTimes()
+			mockOcmInterface.EXPECT().GetTargetCluster(trueClusterId).Return(trueClusterId, testClusterId, nil).AnyTimes()
+			mockOcmInterface.EXPECT().IsClusterHibernating(gomock.Eq(trueClusterId)).Return(false, nil).AnyTimes()
+			mockOcmInterface.EXPECT().GetOCMAccessToken().Return(&testToken, nil).AnyTimes()
+			mockClientUtil.EXPECT().MakeRawBackplaneAPIClientWithAccessToken(backplaneAPIUri, testToken).Return(mockClient, nil).AnyTimes()
+			mockClient.EXPECT().LoginCluster(gomock.Any(), gomock.Eq(trueClusterId)).Return(fakeResp, nil).AnyTimes()
+
+			err := bpSession.RunCommand(cmd, []string{})
+			Expect(err).To(BeNil())
+			Expect(bpSession.Path).Should(ContainSubstring(testClusterId))
+		})
+
+		It("should contains cluster env vars ", func() {
```

Comment:
I still have trouble on this one.
If the KUBECONFIG is set in my `.bashrc`, it will use the value in the new created session as well.
But I think the KUBECONFIG in the session should point to the config file which we created under the session temp dir.
```
$ cat ~/.bashrc | grep KUBEC
export KUBECONFIG=./kubeconfig

$ ocm backplane2 session 23q7udieb4p7vh5o6m4snps24t8496ci
....
Switching to Backplane session 23q7udieb4p7vh5o6m4snps24t8496ci

$ env | grep KUBEC
KUBECONFIG=./kubeconfig
```
The test here checks the EnvVar has value, but did not check the value itself?

Reply:
I can't reproduce this issue and updated the unit test to check the `ocEnv` file containing the KUBECONFIG path. 

https://github.com/openshift/backplane-cli/blob/2ed3988620422532b4ee2716d112627e36890e06/pkg/cli/session/session_test.go#L186